### ISSUE TITLE
Difference between back end and front end rendering using query loop

### ISSUE THREAD

**Issue by phoopee3** [GatherPress/gatherpress#135]
When using the query loop to display events, the date is not rendered correctly (it shows the post date, not the event date), and there is no excerpt show.

![gatherpress-backend-query-loop](https://user-images.githubusercontent.com/418551/187766635-e8b660fd-242d-4b32-be7f-ba33ce4d91e2.png)

On the front end it shows the event date correctly (no time, but that's ok for now i think), and it shows the excerpt.

![gatherpress-frontend-query-loop](https://user-images.githubusercontent.com/418551/187766637-5d48c4d5-7160-44f6-8919-efd65721fb98.png)

**Comment by pbrocks** [comment]
@phoopee3 Can you pull the latest version and check that this has been taken care of? There is a [checkbox in settings](https://github.com/GatherPress/gatherpress/pull/220) that allows you to choose how you want to display dates.

**Comment by phoopee3** [comment]
Sure I’ll take a look later today

jason

**Comment by phoopee3** [comment]
The query loop doesn't show the event date at all, and I don't know of a way to show it in the loop.

**Comment by pbrocks** [comment]
> The query loop doesn't show the event date at all, and I don't know of a way to show it in the loop.

I don't know what you mean here. I've looked at this using 2020 theme and things mostly work as expected. I do notice that using the Query Loop block, the GatherPress setting for showing dates isn't accounted for in the editor, but frontend works.

Custom Query works as expected
https://
![custom-query-loop](https://user-images.githubusercontent.com/5721313/211844780-b6605ffa-1d7f-4463-864c-21249ad92df2.gif)
take.ms/lsYUS

Query Loop Block
https://take.ms/KPGtX
![query-loop-block](https://user-images.githubusercontent.com/5721313/211844751-9beadf49-9a93-4e91-8c32-15ecbe149d1b.gif)

**Comment by pbrocks** [comment]
```php
add_shortcode('do-custom-event-loop', 'do_custom_event_loop' );
/**
 * [do-custom-event-loop] Shortcode for Events 
 *
 * @return string
 */
function do_custom_event_loop() {
	$settings = GatherPress\Core\Settings::get_instance();
	$use_event_date = ( $settings->get_value( 'gp_general', 'general', 'post_or_event_date' ) ? 'using Event Date' : 'using Post Date' );

	$args = array(
	    'post_status' => 'publish',
	    'post_type' => 'gp_event'
	);
 
	$query = new WP_Query( $args );

	ob_start();
	echo '<h4 style="color:salmon">' . $use_event_date . '</h4>';
	if ( $query->have_posts() ) {

	    echo '<ul class="category posts">';

	        while ( $query->have_posts() ) {
	            $query->the_post();
	            ?>
	            <li <?php post_class( 'left' ); ?>>
	            	<?php echo 'Post ID -- ' .  $query->post->ID . ' = '; ?>
	                <a href="<?php the_permalink(); ?>" title="<?php the_title_attribute(); ?>">
	                    <?php the_title(); ?>
	                </a> | <?php the_date( 'F j, Y' ); 
	                 echo ' |*no*format*| ' . the_date();
	                 echo '<br>|| ' . get_the_date( 'Y F j', $query->post->ID  ); 
	                 echo '<br>|| ' . get_the_time( 'Y F j', $query->post->ID  ); 
	                 echo '<br>||| ' . get_the_date( 'F j, Y', $query->post->ID  ); 
	                 echo '<br>|||| ' .  get_the_time('Y-m-d', $query->post->ID );
	                 ?> 
	            </li>
	            <?php
	        }
	    echo '</ul>';
	}

	wp_reset_postdata();
	$return_value = ob_get_clean();
	return $return_value;
}
```

**Comment by pbrocks** [comment]
@mauteri Seems to be an issue in editor for Query Loop blocks, but I wouldn't dig further into this for v1, unless we get someone with deep knowledge of React blocks.

**Comment by mauteri** [comment]
Query Loop block is probably not great for Events. Best to use the GatherPress block `Events List` or the future calendar block. Query Loop is using published date, which events have but are not relevant. On the front end, we have a setting to override the published date with the event date for events.

There might be a filter we can leverage in Query Loop (or one we can put a PR in for Gutenberg so it honors the filter that exists in the publish date filter), but the ordering is still by publish date, which you wouldn't want with events.

We can leave this open, but I don't think there's anything actionable at this point.

**Comment by mauteri** [comment]
@carstingaxion I believe solved this with his Event Query Loop block variation. If you select `Event` it converts itself to the `Event Query Loop` from what I understand from his quick demo.

**Comment by carstingaxion** [comment]
That is correct. Take a look at the screen capture in my least recent comment at https://github.com/GatherPress/gatherpress/issues/599#issuecomment-2241484597

**Comment by carstingaxion** [comment]
I guess this can be closed now, after #962 is merged.
